### PR TITLE
Apollo preesm & spider support

### PR DESCRIPTION
**Pull request by r-lazcano**
Both PREESM & SPiDER now support Apollo optimizations. This branch should be merged together with apolloSpiderSupport in SPiDER repository, otherwise SPiDER applications will not compile.

### CODE REVIEW

File: plugins/org.preesm.codegen.xtend/src/org/preesm/codegen/xtend/spider/visitor/SpiderCodegen.java [hunk 1]
```diff
@@ -305,6 +310,7 @@ public String generateHeaderCode(final PiGraph pg, final SpiderConfig spiderConf
 
     /* Declare Include Files */
     append("#include <spider.h>\n\n");
+    append("#include <apolloAPI.h>\n\n");
```
Comment (antoine-morvan) [inline plugins/org.preesm.codegen.xtend/src/org/preesm/codegen/xtend/spider/visitor/SpiderCodegen.java]: Isn't this highly likely to generate compile errors if the project does not use appolo ?

Comment (r-lazcano) [reply plugins/org.preesm.codegen.xtend/src/org/preesm/codegen/xtend/spider/visitor/SpiderCodegen.java]: You are right. That was part of a first version and should have been removed. I just removed it.

File: plugins/org.preesm.codegen/src/org/preesm/codegen/xtend/task/CodegenEngine.java [hunk 2]
```diff
@@ -380,4 +386,17 @@ private void format(final IFile iFile) {
...
 
   }
+
+  /**
+   * 
+   * @param apolloFlag
+   *          Enable the use of Apollo for intra-actor optimization
+   */
+  public void registerApollo(String apolloFlag) {
+    if (apolloFlag != null && apolloFlag.equalsIgnoreCase("true")) {
```
Comment (antoine-morvan) [inline plugins/org.preesm.codegen/src/org/preesm/codegen/xtend/task/CodegenEngine.java]: rewrite: this.apolloEnabled = (apolloFlag != null && "true".equalsIgnoreCase(apolloFlag))

File: plugins/org.preesm.codegen.xtend/src/org/preesm/codegen/xtend/spider/visitor/SpiderCodegen.java [hunk 3]
```diff
@@ -942,6 +978,7 @@ private void generateFunctionBody(final AbstractActor aa) {
         switch (param.getDirection()) {
           case IN:
             if (param.isIsConfigurationParameter()) {
+              aa.isLocallyStatic();
```
Comment (antoine-morvan) [inline plugins/org.preesm.codegen.xtend/src/org/preesm/codegen/xtend/spider/visitor/SpiderCodegen.java]: this line does nothing if the result is not used

Comment (r-lazcano) [reply plugins/org.preesm.codegen.xtend/src/org/preesm/codegen/xtend/spider/visitor/SpiderCodegen.java]: Removed

File: plugins/org.preesm.codegen/src/org/preesm/codegen/xtend/task/CodegenEngine.java (hunk 2 above)
Comment (r-lazcano) [reply plugins/org.preesm.codegen/src/org/preesm/codegen/xtend/task/CodegenEngine.java]: Changed

**Pull request merged**